### DISCUSSION TITLE
Missing modules ‘bark_hubert_quantizer’ and ‘audiocraft’

### DISCUSSION THREAD

**Discussion by fizion** [rsxdalv/TTS-WebUI#65] [Q&A]
I'm a noob so a one click install always sound good to my ears. I used the latest 5.0 one and ran it.

The interface starts but in my 'Bark Voice Clone' and ''MusicGen' tab I noticed an exclamation mark. No worries I though, has an install button but that doesn't seem to do anything if I look at the console (on both tabs).

The Bark Voice Clone has this text

```
Failed to load voice clone demo
Error: No module named ‘**bark_hubert_quantizer**’
Please install the requirements_bark_hubert_quantizer.txt file
Please check the console for more information
```

And the MusicGen tab has this text:

```
Failed to load voice clone demo
Error: No module named ‘**audiocraft**’
Please install the requirements_bark_hubert_quantizer.txt file
Please check the console for more information
```

I don't mind just installing everything again but I thought maybe it just needs some flag somewhere. I do have those txt files mentioned and I guess I can copy/paste the lines and run them but not sure if that's the right way.

Since it's a one click install some feedback from me as noob my be helpful as well ;)

**Comment by rsxdalv** [comment]
Hi, sorry that it's a bit confusing right now. You should be able to
install it from the UI. However, and that's the reason why it's no longer
installed by default - these installs can and do fail for some
architectures. For example, audiocraft doesn't appear to prioritize
supporting Macbooks. As for Hubert quantizer - it tends to get built on the
system, so if some build tool is missing it might fail.

Either way, I'm happy to help to get them installed even if the button in
the UI doesn't work.

Also, in this case you don't need to reinstall everything.

**Reply by fizion** [reply]
Thanks for getting back!

I thought, well let's clean install (I'm on Windows 11 btw). I still have the same issue that the buttons don't work. If I look at the console box while installing I noticed the following things (might explain something):

This is a thing that seems to be fixed later on in the script when it says on the same json file 'solving environment: done':

```
Input> A
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
```
In between these errors it says there is a newer version of conda (23.5.0) but I don't think that has to do with the issue.

```
CondaError: Downloaded bytes did not match Content-Length
  url: https://conda.anaconda.org/pytorch/win-64/pytorch-2.0.0-py3.10_cuda11.7_cudnn8_0.tar.bz2
  target_path: G:\tts\installer_files\conda\pkgs\pytorch-2.0.0-py3.10_cuda11.7_cudnn8_0.tar.bz2
  Content-Length: 1253569839
  downloaded bytes: 233960448
```

and

```
Running command git clone --filter=blob:none --quiet https://github.com/suno-ai/bark 'C:\Users\oscar\AppData\Local\Temp\pip-install-uzec_bc_\suno-bark_a75d010b0c8b4d5f9108f38bac955afd'
  WARNING: Did not find branch or tag '7280e4e', assuming revision or ref.
  Running command git checkout -q 7280e4e
  Resolved https://github.com/suno-ai/bark to commit 7280e4e
  Installing build dependencies ... done
```

Just before loading the web ui:

```
Failed to load voice clone demo
No module named 'bark_hubert_quantizer'
Failed to load musicgen demo
No module named 'audiocraft'
```

But that's normal since they are not installed on first run I presume. Those are actually the only things I notice being in error. Don't remember having seen the pytorch error when I first tried to install but not sure if it has anything to do with not being able to install both things using the webgui. The pytorch error might be that the internet got a hiccup since I'm on mobile data although that normally doesn't give any problems and even have a nice 150-200Mbps download most of the time.

If I click the buttons on the page it does starts the count thing but the console doesn't report anything.

I thought it might conflict with stable diffusion which I've installed as well but if I remember correctly I saw you write somewhere the package is like it is to not conflict with other versions/installs and run in it's own environment.

I'd like to puzzle a bit and in the end I got Stable Diffusion running perfectly thought to share my experiences, might help others in the end as well.

**Reply by rsxdalv** [reply]
I'll go over the details a little later, but yes, if you use the one click installers, they download a new python (tiny compared to everything else funnily enough), set up a new environment, and all of the dependencies. It might sound a little crazy but most projects are extremely picky with their versions.

**Reply by rsxdalv** [reply]
Which buttons do not work - all of them? I'm just checking because yesterday I managed to completely upend one of my environments while developing.

Also, the manual way to install the "extensions" is (using cmd_windows.bat console):
`cd tts-generation-webui`
`pip install -r requirements_audiocraft.txt` # for musicgen
`pip install -r requirements_bark_hubert_quantizer.txt` # for voice cloning

Perhaps the install buttons have some problems. I am working on RVC right now and it doesn't want to install from within the UI. After that's done the main installation needs an upgrade, this current multi-part install is more of a bugfix to installation failures.

To cover the other errors:
* `CondaError: Downloaded bytes did not match Content-Length` - never seen this one, if I had issues I would just reinstall (that package), or perhaps conda already fixed itself.`
* `Failed to load voice clone demo`
  `No module named 'bark_hubert_quantizer'`
  `Failed to load musicgen demo`
  `No module named 'audiocraft'`
  The first is the description that the "extension" failed to load, the second is the cause - in this case it shows that this module has (probably never) been installed.
* `WARNING: Did not find branch or tag '7280e4e', assuming revision or ref.` - this is normal, maybe there's a way to avoid it.
* _conda out of date_ - this is also normal, will update eventually.

**Reply by fizion** [reply]
Sweet! That worked quite well. 

The buttons that didn't work were the 'Install' buttons on the 'Bark Voice Clone' and 'MusicGen' pages which had the exclamation mark behind the tab name.

After the pip install for both packages I got an error about tensorboardX but that went away after I did 'pip install tensorboardX'.

Only remaining error I currently get is this one but that might have to do with the pytorch package install that went bad. I must say I noticed that it downloaded much faster than the first time I tried (looking at the download bar). So it's almost like it thought it only needed to download 233960448 bytes instead of 1253569839.

Error:
```
2023-07-09 10:50:14 | WARNING | xformers | A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
2023-07-09 10:50:14 | WARNING | xformers | Triton is not available, some optimizations will not be enabled.
This is just a warning: No module named 'triton'
```
It also gives back this one but I don't think it does any wrong

```
G:\tts\tts-generation-webui\server.py:114: GradioDeprecationWarning: The `enable_queue` parameter has been deprecated. Please use the `.queue()` method instead.
  demo.queue(
Running on local URL:  http://0.0.0.0:7860
```

Now you helped me cure those package things I just start from scratch and check if that pytorch/cuda thing gets installed the right way.

Little question, is it needed to put some directories in an environment path in Windows? I was trying to see if conda and pytorch were installed the right way and thought you could do 'Conda -version' to check it. But if I do so it says it doesn't know the command.

Thanks for your time! For me as an oldie it's quite a miracle what you can do these days with 'just' a desktop computer, well with a beefy GPU that is.

**Reply by rsxdalv** [reply]
I released the v6 installer which, when things go well, will completely avoid the install buttons appearing. If your connection is slow and you don't care for RVC just yet, I recommend not upgrading immediately. I tried doing a partial update but it failed on my system and once again I had to recreate the conda environment, so the only real way to v6 is to install and copy over.

> The buttons that didn't work were the 'Install' buttons on the 'Bark Voice Clone' and 'MusicGen' pages which had the exclamation mark behind the tab name.

Ah, I guess I was too optimistic when I made them, but at least the issue is limited.

> After the pip install for both packages I got an error about tensorboardX but that went away after I did 'pip install tensorboardX'.

This is probably the core issue I'll need to fix, thanks for reporting it!

> Only remaining error I currently get is this one but that might have to do with the pytorch package install that went bad. I must say I noticed that it downloaded much faster than the first time I tried (looking at the download bar). So it's almost like it thought it only needed to download 233960448 bytes instead of 1253569839.
> 
> Error:
> 
> 2023-07-09 10:50:14 | WARNING | xformers | A matching Triton is not available, some optimizations will not be enabled.
> Error caught was: No module named 'triton'
> 2023-07-09 10:50:14 | WARNING | xformers | Triton is not available, some optimizations will not be enabled.
> This is just a warning: No module named 'triton'
> It also gives back this one but I don't think it does any wrong
> 

Yes, triton _kind of doesn't matter_, and I also get that warning.

> G:\tts\tts-generation-webui\server.py:114: GradioDeprecationWarning: The `enable_queue` parameter has been deprecated. Please use the `.queue()` method instead.
>   demo.queue(
> Running on local URL:  http://0.0.0.0:7860
> 

This is small but should be easy to fix the message, will check, thanks!

> Now you helped me cure those package things I just start from scratch and check if that pytorch/cuda thing gets installed the right way.

As for pytorch and cuda - the correctness usually depends on what you are installing. Some apps might require pytorch 2.0.0 others 2.0.1 (which 99% of the time is insignificant, but if a package says it needs 2.0.0 it might even uninstall 2.0.1 to install a 2.0.0 version and then break everything - this happened to me with RVC). Currently this project aligns with pytorch 2.0.0 and cuda 11.7 + NVidia GPU, with cpu and some MPS support.

>Little question, is it needed to put some directories in an environment path in Windows? I was trying to see if conda and pytorch were installed the right way and thought you could do 'Conda -version' to check it. But if I do so it says it doesn't know the command.

Conda is not installed in path, the 'start_*' functions:
* install conda
* run it
* use the now installed conda

to access them, such as `conda -version` or `pip show torch` using the cmd_ is easy, because it runs all the commands to put the local conda in "path" (only for the terminal session) and activate it.

> Thanks for your time! For me as an oldie it's quite a miracle what you can do these days with 'just' a desktop computer, well with a beefy GPU that is.

Thank you too, this feedback is useful and welcome! I am also surprised at how much can be done even with 'outdated for gaming' GPUs.
